git/internal/e2e/run.sh

```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+# This script runs e2e tests for pkg/git.
+
+set -o errexit
+PROJECT_DIR=$(git rev-parse --show-toplevel)
+DIR="$(cd "$(dirname "$0")" && pwd)"
+
+cd "$PROJECT_DIR"/git/libgit2 && make libgit2
+
+source "$DIR"/setup_gitlab.sh
+
+LIBGIT2_IMG=ghcr.io/fluxcd/golang-with-libgit2
+LIBGIT2_TAG=v0.2.0
```

Comment:
On SC we ensure that libgit2 tags are always aligned across the different files that have it set.
We should do the same on pkg, either on this or on a follow-up PR.

https://github.com/fluxcd/source-controller/blob/09ef651bebcd853df555eb689ba69a2cea33b21c/Makefile#L199-L211